robotology-superbuild on Unstable branches with ROBOTOLOGY_ENABLE_DYNAMICS ON and ROBOTOLOGY_ENABLE_DYNAMICS_FULL_DEPS OFF compilation broken

In the latest nightly build of the robotology-superbuild (https://github.com/robotology/robotology-superbuild/runs/1752668669), a few jobs failed with this error:
~~~
2021-01-23T03:21:56.2441183Z CMake Error at cmake/AddBipedalLocomotionApplication.cmake:30 (add_executable):
2021-01-23T03:21:56.2442530Z   Target "joint-trajectory-player" links to target
2021-01-23T03:21:56.2443460Z   "BipedalLocomotion::Planners" but the target was not found.  Perhaps a
2021-01-23T03:21:56.2444324Z   find_package() call is missing for an IMPORTED target, or an ALIAS target
2021-01-23T03:21:56.2444923Z   is missing?
2021-01-23T03:21:56.2445370Z Call Stack (most recent call first):
2021-01-23T03:21:56.2446490Z   utilities/joint-trajectory-player/CMakeLists.txt:5 (add_bipedal_locomotion_application)
2021-01-23T03:21:56.2447342Z 
2021-01-23T03:21:56.2447567Z 
~~~

At a first glance, all this jobs that failed with this error had ROBOTOLOGY_ENABLE_DYNAMICS ON and ROBOTOLOGY_ENABLE_DYNAMICS_FULL_DEPS OFF .

--- a/cmake/BipedalLocomotionFrameworkFindDependencies.cmake
+++ b/cmake/BipedalLocomotionFrameworkFindDependencies.cmake
@@ -233,3 +233,6 @@ framework_dependent_option(FRAMEWORK_COMPILE_MasImuTest
   "Compile test on the MAS IMU?" ON
   "FRAMEWORK_COMPILE_YarpImplementation;FRAMEWORK_COMPILE_matioCppConversions" OFF)
 
+framework_dependent_option(FRAMEWORK_COMPILE_JointTrajectoryPlayer
+  "Compile joint-trajectory-player application?" ON
+  "FRAMEWORK_COMPILE_YarpImplementation;FRAMEWORK_COMPILE_Planners;FRAMEWORK_COMPILE_RobotInterface;FRAMEWORK_COMPILE_matioCppConversions;FRAMEWORK_USE_matioCpp;FRAMEWORK_USE_YARP" OFF)

--- a/utilities/joint-trajectory-player/CMakeLists.txt
+++ b/utilities/joint-trajectory-player/CMakeLists.txt
@@ -2,16 +2,20 @@
 # GNU Lesser General Public License v2.1 or any later version.
 # Authors: Ines Sorrentino
 
-add_bipedal_locomotion_application(
-  NAME joint-trajectory-player
-  SOURCES src/Main.cpp src/Module.cpp
-  HEADERS include/BipedalLocomotion/JointTrajectoryPlayer/Module.h
-  LINK_LIBRARIES  YARP::YARP_dev 
-                  BipedalLocomotion::Planners 
-                  BipedalLocomotion::ParametersHandlerYarpImplementation 
-                  BipedalLocomotion::RobotInterfaceYarpImplementation 
-                  matioCpp::matioCpp 
-                  BipedalLocomotion::matioCppConversions
-  )
+if(FRAMEWORK_COMPILE_JointTrajectoryPlayer)
 
-install_ini_files(${CMAKE_CURRENT_SOURCE_DIR}/config)
+  add_bipedal_locomotion_application(
+    NAME joint-trajectory-player
+    SOURCES src/Main.cpp src/Module.cpp
+    HEADERS include/BipedalLocomotion/JointTrajectoryPlayer/Module.h
+    LINK_LIBRARIES  YARP::YARP_dev
+    BipedalLocomotion::Planners
+    BipedalLocomotion::ParametersHandlerYarpImplementation
+    BipedalLocomotion::RobotInterfaceYarpImplementation
+    matioCpp::matioCpp
+    BipedalLocomotion::matioCppConversions
+    )
+
+  install_ini_files(${CMAKE_CURRENT_SOURCE_DIR}/config)
+
+endif()

